packages/x-components/src/components/modals/base-id-modal-open.vue

```diff
@@ -1,36 +1,39 @@
 <template>
-  <BaseEventButton
-    v-on="$listeners"
-    :events="events"
-    class="x-button x-events-modal-id-open-button"
-    data-test="open-modal-id"
-  >
-    <!-- @slot (Required) Button content with a text, an icon or both -->
-    <slot />
-  </BaseEventButton>
+  <NoElement v-on="$listeners" data-test="open-modal-id">
+    <!--
+      @slot opening-element. It's the element that will trigger the modal opening. It's a
+      button by default.
+        @binding {Function} openModal - The function to open the modal.
+    -->
+    <slot :openModal="emitOpenModalEvent" name="opening-element">
+      <button @click="emitOpenModalEvent" class="x-button x-events-modal-id-open-button">
+        <slot />
+      </button>
+    </slot>
+  </NoElement>
 </template>
 
 <script lang="ts">
   import Vue from 'vue';
   import { Component, Prop } from 'vue-property-decorator';
-  import { XEventsTypes } from '../../wiring/events.types';
-  import BaseEventButton from '../base-event-button.vue';
+  import { NoElement } from '../no-element';
 
   /**
-   * Component containing an event button that emits {@link XEventsTypes.UserClickedOpenModal} when
-   * clicked with the modalId as payload. It has a default slot to customize its contents.
+   * Component that allows to open a modal by emitting {@link XEventsTypes.UserClickedOpenModal}
+   * with the modalId as payload. It's fully customizable as it exposes the opening event but by
+   * default it renders a customizable button.
    *
    * @public
    */
   @Component({
-    components: { BaseEventButton }
+    components: { NoElement }
   })
   export default class BaseIdModalOpen extends Vue {
     @Prop({ required: true })
     protected modalId!: string;
 
-    protected get events(): Partial<XEventsTypes> {
-      return { UserClickedOpenModal: this.modalId };
+    protected emitOpenModalEvent(): void {
+      this.$x.emit('UserClickedOpenModal', this.modalId, { target: this.$el as HTMLElement });
```

Comment:
Not really relevant, as we don't use it, but should we use `event.target` as the `target` metadata field here? If you replace the template, the target of the even't might not be the wrapper.